common/lib/xmodule/xmodule/docs/decisions/0001-move-due-dates-to-db.rst

```diff
@@ -0,0 +1,52 @@
+1. Move Due Dates to Relational Database
+----------------------------------------
+
+Status
+------
+
+Proposed
+
+Context
+-------
+
+We want an authoritative and readily-accessible place to track due dates for sections/units in a course.  We also want to override due dates per learner, and record the reason for the override. The current implementation stores due dates in the XBlock (in mongodb). If you want to know all of the due dates in a course, you have to walk the entire course structure and pull out each date, which is an inefficient operation.
```

Comment:
BTW, while I completely agree with the long term goal here, please keep in mind that a major roadblock to this will be access permissions. Even if we have we have start/due dates and overrides neatly placed in a sane data model for querying, we won't necessarily be able to tell a user "here's all the things you have due" without potentially expensive `has_access` calls.

I still think what you're proposing is absolutely worth doing. It gives us the data model to implement the individual due date overrides + auditing, and allows for better querying of those things (e.g. "What are all the extensions I've given for this course?") It puts us on a path towards being able to eventually doing these types of queries. We might even be able to cheat a little on the `has_access` calls to make them faster. But I just don't want anyone to think that making this data model makes such a student-facing dashboard a trivial addition. It's necessary to get us there, but likely not sufficient by itself.